adserver/models.py

```diff
@@ -1106,15 +1127,20 @@ def get_absolute_url(self):
             },
         )
 
-    def incr(
-        self, impression_type, publisher, request=None, div_id=None, ad_type_slug=None
-    ):
-        """Add to the number of times this action has been performed, stored in the DB."""
+    def incr(self, impression_type, publisher):
+        """
+        Add to the number of times this action has been performed, stored in the DB.
+
+        TODO: Refactor this method, moving it off the Advertisement class since it can be called
+              without an advertisement when we have a Decision and no Offer.
```

Comment:
This is pretty awkward. I wasn't even sure calling this method without an instance was valid. Would making this a `classmethod` be better?

Reply:
Probably. I think probably moving it to the Publisher is more correct, since that is never null. Didn't seem like a huge priority right now tho, and would have been even more refactoring in the PR.

Reply:
I tried to refactor it, and it still leads to some janky logic like `publisher.incr(impression_type=CLICKS, advertisement=self)` -- doesn't seem much better.